SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -1,271 +1,215 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
 #include "EngineClasses/SpatialLoadBalanceEnforcer.h"
+
 #include "EngineClasses/SpatialVirtualWorkerTranslator.h"
 #include "Schema/AuthorityIntent.h"
 #include "Schema/Component.h"
 #include "Schema/ComponentPresence.h"
 #include "Schema/NetOwningClientWorker.h"
 #include "SpatialCommonTypes.h"
+#include "SpatialConstants.h"
 #include "SpatialGDKSettings.h"
 
 DEFINE_LOG_CATEGORY(LogSpatialLoadBalanceEnforcer);
 
-using namespace SpatialGDK;
-
-SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName& InWorkerId,
-													   const USpatialStaticComponentView* InStaticComponentView,
-													   const SpatialVirtualWorkerTranslator* InVirtualWorkerTranslator)
+namespace SpatialGDK
+{
+SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName& InWorkerId, const FSubView& InSubView,
+													   const SpatialVirtualWorkerTranslator* InVirtualWorkerTranslator,
+													   TUniqueFunction<void(EntityComponentUpdate)> InUpdateSender)
 	: WorkerId(InWorkerId)
-	, StaticComponentView(InStaticComponentView)
+	, SubView(&InSubView)
 	, VirtualWorkerTranslator(InVirtualWorkerTranslator)
+	, UpdateSender(MoveTemp(InUpdateSender))
```

Comment:
Looks as if we do not queue up the acl changes anymore and synchronously make component updates. Is there a specific reason for this?

Reply:
Ahh it looks like we do queue stuff up but can bypass that queue. Ok, so how come we need to bypass the queue?

Reply:
Yep! the queue was racy and complicated and made testing this hard. The idea behind the LB enforcer is actually to _async_ read LB component updates, and feed those back into Spatial ASAP. The queue was a "performance improvement" to minimise number of ACL updates, which in reality didn't really happen. 

In order to make this actually asynchronously update the ACL as fast as possible, this PR does two things:
 - Introduces sending it off immediately after its been refreshed, as you see here- avoids fun bookkeeping of "did I lose ACL authority since I queued this?"
 - Moves LB enforcer logic to before actor processing logic in the tick- short circuiting still happens as soon as we calculate it, but we want the LB enforcer to process any new potential ACL changing updates from other workers ASAP

This should be a performance improvement overall for handover, actually, although the main idea is just to make it easier to reason about.

Reply:
> Ahh it looks like we do queue stuff up but can bypass that queue. Ok, so how come we need to bypass the queue?

We don't queue stuff anymore- we update our local state, but there's no queue of ACL updates

Reply:
Right yes indeed. But we still send off a lot of acl updates at the end of TickDispatch, the same point before. Some have been sped up though by using ShortCircuitMaybeRefreshAcl right?

Reply:
yeah, I didn't follow what you meant. We now flush early immediately after the LB advance (as of a recent commit), but short circuit refreshes only get flushed with the rest of the updates. This has to happen to make sure we don't end up with auth intent and ACLs not matching in spatial anyway for the duration of a tick